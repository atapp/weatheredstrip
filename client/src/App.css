a {
  text-decoration: none;
}

html,
body {
  margin: 0;
  padding: 0;
  height: 100%;
  background-color: #282c34;
}

time {
  display: block;
}

.app {
}

.app-header {
  background-color: #282c34;
  height: calc(40px + 2vmin);
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.app-header-content {
  width: 90%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

#app-info {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: left;
  height: 100%;
}

#app-logo {
  width: calc((2vmin + 30px) * 350 / 240);
  height: auto;
}

.app-title {
  font-size: calc(30px + 2vmin);
  min-width: 315px;
  color: white;
  text-align: left;
}

.button {
  display: inline-block;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: 0 0.25rem;
  font-size: 0.9375rem;
  line-height: 1.50;
  border-radius: 3px;
  outline: none;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.button:disabled {
  background-color: #A0A8B0;
  border-color: #A0A8B0;
}

.button-selection {
  margin-bottom: 5px;
  width: 100%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: left;
  max-width: 500px;
}

.primary {
  color: #fff;
  background-color: #467fcf;
  border-color: #467fcf;
}

.primary:hover {
  background-color: #6099E9;
  border-color: #6099E9;
}

.secondary {
  color: #fff;
  background-color: #868e96;
  border-color: #868e96;
}

.secondary:hover {
  background-color: #A0A8B0;
  border-color: #A0A8B0;
}

.content {
  /* width: 90%; */
  min-width: 440px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: auto;
  padding-bottom: 40px;
  background-color: #fff;
}

.content-header {
  display: flex;
  flex-direction: row;
}

.dot-loader{
height: 5px;
width: 5px;
border-radius: 50%;
background-color: #282c34;
position: relative;
-webkit-animation: 1.2s scaleDown ease-in-out infinite ;
animation: 1.2s scaleDown ease-in-out infinite ;
}

.dot-loader:nth-child(2){
margin: 0 15px;
-webkit-animation: 1.2s scaleDown ease-in-out infinite .15555s;
animation: 1.2s scaleDown ease-in-out infinite .15555s;
}

.dot-loader:nth-child(3){
-webkit-animation: 1.2s scaleDown ease-in-out infinite .300000s;
animation: 1.2s scaleDown ease-in-out infinite .300000s;
}

@-webkit-keyframes scaleDown{
0%, 80%, 100%{
-webkit-transform: scale(0);
        transform: scale(0);
}
40%{
-webkit-transform: scale(1);
        transform: scale(1);
}
}

@keyframes scaleDown{
0%, 80%, 100%{
-webkit-transform: scale(0);
        transform: scale(0);
}
40%{
-webkit-transform: scale(1);
        transform: scale(1);
}
}

.footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  min-height: 30px;
  padding: 5px 0px;
  color: #fff;
  font-size: 0.7em;
  background-color: #282c34;
}

.footer a {
  color: #fff;
}

.footer a:hover {
  color: #DDD;
}

.footer-content {
  margin: auto;
  width: 90%;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  text-align: center;
}

h1 {
  border-bottom: 1px #DDD solid;
  margin-bottom: 0;
}

h2 {
  margin-bottom: 0;
}

h3 {
  margin-bottom: 0;
}

.lds-ring {
  display: block;
  position: relative;
  margin: 50px auto ;
  width: 95px;
  height: 95px;
}

.lds-ring div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: 80px;
  height: 80px;
  margin: 6px;
  border: 6px solid #6CA5F5;
  border-radius: 50%;
  animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border-color: #6CA5F5 transparent transparent transparent;
}

.lds-ring div:nth-child(1) {
  animation-delay: -0.45s;
}

.lds-ring div:nth-child(2) {
  animation-delay: -0.3s;
}

.lds-ring div:nth-child(3) {
  animation-delay: -0.15s;
}

@keyframes lds-ring {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.margin-6225 {
  margin-left: 62.25px;
}

.metar-rvr {
  flex: 1;
  display: flex;
  padding-left: 10px;
  flex-direction: column;
  justify-content: flex-start;
}

#navbar {
  flex: 100px 0 0;
  flex-shrink: 0;
  border-right: 1px #D1D1D1 solid;
  background-color: #F1F1F1;
}

.navbar-content {
  position: sticky;
  top: 0px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}

#notam-header {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

#notams {
  flex: 1;
  padding-right: 10px;
  display: flex;
  flex-direction: column;
}

.notif-text {
  font-size: .9em;
  font-family: 'Source Code Pro', monospace;
  text-align: left;
  margin-bottom: 12px;
}

p {
  margin-bottom: 0;
}

#root {
  position: relative;
  min-height: 100%;
}

#rvr {
  display: flex;
  flex-shrink: 0;
  max-height: 300px;
  max-width: 300px;
  margin: 0 auto;
}

.searchbox {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 75vmin;
  max-width: 300px;
  min-width: 200px;
  max-height: 30px;
}

.searchbox-input{
  font-size: 16px;
  width: 60%;
  border-radius: 3px;
  border: none;
  padding: 5px;
}

.searchbox-button {
  background-color: #EEE;
  width: 30%;
  font-size: 14px;
  border-radius: 5px;
  border: none;
}

.searchbox-button:hover {
  background-color: #FFF;
}

.searchbox-button:active {
  background-color: #DDD;
}

.selected-content {
  display: flex;
  flex-direction: row-reverse;
}

.station {
  margin: 0;
  left: 0px;
  border: none;
  border-left: 4px #F1F1F1 solid;
  border-bottom: 1px rgb(204, 204, 204) solid;
  padding: 0 10px;
  font-family: 'Mukta', sans-serif;
  font-size: 1.8rem;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  position: relative;
  color: #444;
  background-color: #F1F1F1;
  z-index: 1;
}

.station:hover {
  background-color: #FCFCFC;
  color: #467fcf;
  border-left: 4px #FCFCFC solid;
}

.station-selected {
  left: 0px;
  border-left: 4px #467fcf solid;
  color: #467fcf;
  background-color: #FFF;
  vertical-align: center;
  text-align: center;
  z-index: 2;
}

.station-selected:hover {
  background-color: #FFF;
  border-left: 4px #467fcf solid;
}

.station-name {
  font-family: 'Mukta', sans-serif;
  font-size: 40px;
  color: #333;
}

.subtitle {
  text-align: left;
  font-family: 'Mukta', sans-serif;
  font-size: 28px;
  color: #333;
}

.timestamp {
  margin: auto;
  position: relative;
  right: 5%;
  display: flex;
  flex-direction: row;
  z-index: 0;
}

.timestamp-names {
  text-align: right;
  margin-right: 10px;
}

.timestamp-time {
  text-align: left;
}

.user-content {
  flex: 100% 1 1;
  display: flex;
  background-color: #FFF;
  flex-direction: column;
  align-content: center;
  padding: 0 10px;
  min-height: calc(100vh - 40px - 10px - 55px);
  padding-bottom: 10px;
}

@media only screen and (max-width: 800px) {
  .metar-rvr {
    flex: auto;
    padding-left: 0;
    flex-direction: column;
  }
  .selected-content {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
  }
  #rvr {
    margin: auto;
  }
  #notams {
    flex: auto;
    padding-right: 0;
  }
  .user-content {
    min-height: calc(100vh - 40px - 10px - 90px)
  }
}

@media only screen and (max-width: 700px) {
  .app-header {
    height: calc(12vmax);
  }

  .app-title {
    min-width: 0;
    width: 100%;
  }

  #app-info {
    justify-content: center;
    height: calc(5vmax);
  }

  .app-header-content {
    height: 100%;
    width: 100%;
    flex-direction: column;
    justify-content: space-evenly;
  }
}

@media only screen and (max-width: 550px) {
  .timestamp {
    margin: auto;
    position: inherit;
  }

  .content-header {
    display: flex;
    flex-direction: column;
  }
}

@media only screen and (max-width: 450px) {
  body, html {
    background-color: #282c34
  }

  #app-logo {
    height: 100%;
  }

  .app-title {
    display: none;
  }

  .button-selection {
    width: 100%;
  }

  .content {
    background-color: #FFF;
    flex-direction: column;
    width: 100%;
    min-width: inherit;
    padding-bottom: 74px;
  }

  .footer-content {
    margin: auto;
    width: 90%;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
  }

  .footer-item {
    margin: 2px 0px;
  }

  .margin-6225 {
    margin-left: 0;
  }

  #navbar {
    position: sticky;
    top: 0;
    padding-top: 0;
    border: none;
    margin-bottom: 5px;
    flex: 100% 1 1;
    overflow: visible;
    white-space: nowrap;
    justify-content: left;
    z-index: 100;
  }

  .navbar-content {
    display: flex;
    flex-direction: row;
  }

  #notam-header {
    background-color: #FFF;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    position: sticky;
    top: 32px;
    z-index: 50;
  }

  #rvr {
    flex: none;
    max-width: 100%;
    max-height: 100%;
  }

  .station {
    left: 0px;
    flex: 1 0 80px;
    border: none;
    font-size: 1.2rem;
    border-bottom: 1px #D1D1D1 solid;
    border-top-left-radius: 0px;
    border-bottom-left-radius: 0px;
  }

  .station-name {
    display: none;
  }

  .station-selected {
    font-weight: bold;
    border-bottom: 1px #467fcf solid;
    border: 0 0 2px 0 black solid;
    -moz-box-shadow:    none;
    -webkit-box-shadow: none;
    box-shadow:         none;
  }

  .subtitle {
    background-color: #FFF;
    position: sticky;
    top: 32px;
    z-index: 50;
  }

  .timestamp {
    margin: auto;
    position: inherit;
    z-index: auto;
  }

  .user-content {
    border: none;
    min-height: calc(100vh - 74px - 37px - 10px - 90px)
  }
}
