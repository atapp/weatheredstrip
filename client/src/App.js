import React, { Component } from 'react';
import './App.css';


// Import of Components
import { Header, Query } from './components';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Query url={this.props.url} />
      </div>
    );
  }
}

export default App;
