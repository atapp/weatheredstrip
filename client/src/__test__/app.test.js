import React from 'react';
import { render, cleanup } from '@testing-library/react'
import App from '../App';
import 'jest-dom/extend-expect';
afterEach(cleanup)

it('Renders nothing is provided', () => {
  const { container } = render(<App/>)
  const app = container.firstChild
  expect(app).toMatchSnapshot()
});
