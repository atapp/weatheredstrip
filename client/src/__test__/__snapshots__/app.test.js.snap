// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders nothing is provided 1`] = `
<div
  class="app"
>
  <header
    class="app-header"
  >
    <div
      class="app-header-content"
    >
      <a
        href="/"
      >
        <div
          id="app-info"
        >
          <img
            alt="App Logo"
            id="app-logo"
            src="logo.svg"
          />
          <div
            class="app-title"
          >
            Weathered Strip
          </div>
        </div>
      </a>
      <form
        class="searchbox"
      >
        <input
          class="searchbox-input"
          placeholder="CYMX CYUL..."
          type="text"
          value=""
        />
        <button
          class="searchbox-button"
          type="submit"
        >
          Get Info
        </button>
      </form>
    </div>
  </header>
  <div
    class="content"
  >
    <div
      class="user-content fullborder"
    >
      <h1>
        Welcome to Weathered Strip
      </h1>
      <p>
        Weathered Strip was created to help gather all required flight planning documentation in one place. It provides NOTAMs, METARs/TAFs and live RVR for each requested airport. More will be included later.
      </p>
      <h2>
        How to...
      </h2>
      <h3>
        First use
      </h3>
      <p>
        To use Weathered Strip, please enter the ICAO airport codes of each airport you'd like to assess in the seachbox located in the header.
      </p>
      <p>
        Note: for multiple ICAO airport codes, separate them by either spaces or commas.
      </p>
      <h3>
        Report an issue
      </h3>
      <p>
        Weathered Strip is not perfect and its developer is always looking to improve its feel and squash nasty bugs which create unexpected behaviours.
      </p>
      <p>
        Please use the "Report an Issue" link located in the footer to request a new feature, make a suggestion, report a bug, etc.
      </p>
      <h3>
        Contribute
      </h3>
      <p>
        Want to contribute to Weathered Strip? Please do! Fork me on 
        <a
          href="https://github.com/GregoryHamel/weatheredstrip/"
        >
          GitHub‚Ñ¢
        </a>
         and make a pull request. I'll be more than happy to review it and include it in the code.
      </p>
    </div>
  </div>
  <div
    class="footer"
  >
    <div
      class="footer-content"
    >
      <div
        class="footer-item"
      >
        Data from 
        <a
          href="https://flightplanning.navcanada.ca"
        >
          NAV CANADA
        </a>
        ‚Ñ†, 
        <a
          href="https://www.aviationweather.gov"
        >
          AWC
        </a>
        ‚Ñ† and 
        <a
          href="https://notams.aim.faa.gov"
        >
          FAA
        </a>
        ‚Ñ†
      </div>
      <div
        class="footer-item"
      >
        <div>
          <a
            href="https://github.com/GregoryHamel/weatheredstrip/issues"
          >
            Report an Issue
          </a>
        </div>
        <div>
          ¬© 
          <a
            href="http://www.greghamel.com"
          >
            Greg Hamel - 
            2019
          </a>
        </div>
      </div>
      <div
        class="footer-item"
      >
        Made in 
        <span
          aria-label="Canada"
          role="img"
        >
          üçÅ
        </span>
         with 
        <span
          aria-label="love"
          role="img"
        >
          ‚ù§Ô∏è
        </span>
      </div>
    </div>
  </div>
</div>
`;
